cpp/src/arrow/c/bridge_test.cc

```diff
@@ -4506,6 +4554,29 @@ class TestArrayStreamRoundtrip : public BaseArrayStreamTest {
     ASSERT_TRUE(weak_reader.expired());
   }
 
+  void Roundtrip(std::shared_ptr<ChunkedArray> src,
+                 std::function<void(const std::shared_ptr<ChunkedArray>&)> check_func) {
+    ArrowArrayStream c_stream;
+
+    // One original copy which to compare the result, one copy held by the stream
+    std::weak_ptr<ChunkedArray> weak_src(src);
+    ASSERT_EQ(weak_src.use_count(), 2);
```

Comment:
Instead of hardcoding the expected ref count, why not just remember it and then compare it later?